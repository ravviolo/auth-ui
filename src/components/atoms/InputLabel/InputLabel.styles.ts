import styled from 'styled-components';

export const Label = styled.label`
  font-size: 0.9rem;
`;
