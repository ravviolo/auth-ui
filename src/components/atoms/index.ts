export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
