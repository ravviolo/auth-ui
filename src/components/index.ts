export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
