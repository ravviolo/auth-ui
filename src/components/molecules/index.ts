export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
