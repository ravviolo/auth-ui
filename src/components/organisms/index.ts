export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
