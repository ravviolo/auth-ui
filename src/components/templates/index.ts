export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
