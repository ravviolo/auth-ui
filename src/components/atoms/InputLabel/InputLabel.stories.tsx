import { ComponentMeta, ComponentStory } from '@storybook/react';

import { InputLabel } from 'components/atoms';

export default {
  title: 'Atoms/InputLabel',
  component: InputLabel,
  args: {
    htmlFor: 'inputFieldId',
    label: 'Test Label',
  },
} as ComponentMeta<typeof InputLabel>;

export const Default: ComponentStory<typeof InputLabel> = (args) => (
  <InputLabel {...args} />
);
Default.storyName = 'InputLabel';
