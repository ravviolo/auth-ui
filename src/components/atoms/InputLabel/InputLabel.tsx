import * as S from './InputLabel.styles';

export interface Props {
  label: string;
  htmlFor: string;
  testId: string;
}

export const InputLabel = ({ htmlFor, label, testId }: Props) => {
  return (
    <S.Label data-testid={testId} htmlFor={htmlFor}>
      {label}
    </S.Label>
  );
};
