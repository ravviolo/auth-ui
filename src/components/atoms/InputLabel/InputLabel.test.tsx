import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { InputLabel } from 'components/atoms';

import { Props as InputLabelProps } from './InputLabel';

describe('InputLabel', () => {
  const props: InputLabelProps = {
    htmlFor: 'test-input-label',
    label: 'Test Input',
    testId: 'test-input-label',
  };

  it('should display label text received via props', () => {
    render(<InputLabel {...props} />);

    const inputLabel = screen.getByTestId('test-input-label');

    expect(inputLabel).toHaveTextContent('Test Input');
  });

  it("should correctly assign 'for' attribute received via props", () => {
    render(<InputLabel {...props} />);

    const inputLabel = screen.getByTestId('test-input-label');

    expect(inputLabel).toHaveAttribute('for', 'test-input-label');
  });

  it('should match snapshot', () => {
    const tree = create(<InputLabel {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
