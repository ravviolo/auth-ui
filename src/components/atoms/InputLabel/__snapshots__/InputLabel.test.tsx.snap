// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel matches snapshot 1`] = `
<label
  className="sc-papXJ daxRFt"
  data-testid="test-input-label"
  htmlFor="test-input-label"
>
  Test Input
</label>
`;
